functions-app/report-psychometrician/service/detections/detect-check-took-too-long.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const RA = require('ramda-adjunct')
+const R = require('ramda')
+
+const getCheckStartedDate = require('./get-check-started-date')
+const getCheckCompleteDate = require('./get-check-complete-date')
+const report = require('./report')
+
+const checkTookTooLong = function (data) {
+  if (!RA.isPlainObj(data)) {
+    throw new TypeError('data should be an object')
+  }
+
+  const markedAnswers = R.pathOr([], ['markedAnswers', 'answer'], data) // contains the question info
+  const config = R.pathOr({}, ['checkPayload', 'config'], data)
+  const numberOfQuestions = R.length(markedAnswers) || 0
+
+  // Calculate the max total time allowed for the check
+  const maxCheckSeconds = (numberOfQuestions * config.loadingTime) +
+    (numberOfQuestions * config.questionTime) +
+    (config.questionReader ? numberOfQuestions * 2.5 : 0)
+
+  const checkStartedDate = getCheckStartedDate(
+    null, // always get the date from the audit log
+    R.prop('checkPayload', data)
```

Comment:
shouldn't that be R.pathOr to provide a default if `checkPayload` isn't present?

Reply:
If the payload is not present the implementation of getCheckStartedDate will work correctly.  In fact, passing in undefined makes it easier to check than an empty object.